src/VirtoCommerce.ElasticAppSearch.Data/Services/Converters/FieldNameConverter.cs

```diff
@@ -19,6 +19,9 @@ public virtual string ToProviderFieldName(string indexFieldName)
         // Only lowercase letters allowed
         var providerFieldName = indexFieldName.ToLowerInvariant();
 
+        // Replase whitespaces with underscores
+        providerFieldName = Regex.Replace(providerFieldName, @"\W", "_");

```

Comment:
I intoroduces `Replacements` to make changes reversible, i.e. to restore original field name in `ToIndexFieldName`, but you may ignore it if it's ok for VC